ea-factories/src/factories/balance.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  ExecuteFactory,
+  Config,
+  ResponseData,
+  Account,
+  SequenceResponseData,
+} from '@chainlink/types'
+import objectPath from 'object-path'
+import { Requester, Validator, AdapterError } from '@chainlink/external-adapter'
+
+const DEFAULT_DATA_PATH = 'addresses'
+const DEFAULT_CONFIRMATIONS = 6
+
+const WARNING_NO_OPERATION = 'No Operation: unsupported'
+const ERROR_NO_OPERATION_MISSING_ADDRESS = 'No Operation: address param is missing'
+
+export type IsSupported = (coin: string, chain: string) => boolean
+export type GetBalance = (account: Account, config: BalanceConfig) => Promise<ResponseData>
+export type GetBatchBalance = (
+  accGroup: [string, AccountGroup],
+  config: Config,
+) => Promise<ResponseData>
```

Comment:
My suggestion would be to have a generic type that will be a response to our data requests. It will carry some kind a result (an answer to our data request) and a payload or some underlying data that the result was made from:

```javascript
export type DataResponse<R, P> = {
  result: R,
  payload: P
}
```

Then we can declare these two type of requests for data:

```suggestion
export type GetBalance = (account: Account, config: BalanceConfig) => Promise<DataResponse<Account, any>>
export type GetBalances = (accounts: Account[], config: BalanceConfig) => Promise<DataResponse<Account[], any>>
```